lib/std/private/jsutils.nim

```diff
@@ -0,0 +1,12 @@
+when defined(js):
+  type
+    ArrayBuffer* = ref object of JsRoot
+    Float64Array* = ref object of JsRoot
+    Uint32Array* = ref object of JsRoot
+
+  func newArrayBuffer*(n: int): ArrayBuffer {.importjs: "new ArrayBuffer(#)".}
+  func newFloat64Array*(buffer: ArrayBuffer): Float64Array {.importjs: "new Float64Array(#)".}
+  func newUint32Array*(buffer: ArrayBuffer): Uint32Array {.importjs: "new Uint32Array(#)".}
+
+  func `[]`*(arr: Uint32Array, i: int): uint32 {.importjs: "#[#]".}
+  func `[]=`*(arr: Float64Array, i: int, v: float) {.importjs: "#[#] = #".}
```

Comment:
should this be `arr: Float64Array` or `arr: var Float64Array` ? I'm really not sure here since the type is `ref`, but really, Float64Array + friends act like seq, so var seems to make more sense here.

Reply:
`var` will generates array of typed array

Reply:
yes, this is generally true of `var` in `nim js`; the question is whether this has a performance impact in this case or whether v8 can optimize this (I'm not sure how to check other than performance timing)

Reply:
It seems to cause problems, but I forgot the example.

Reply:
note that for var you need the awkard syntax like this: `[#][0][0]`
```nim
# eg: jsbigints:
func inc*(this: var JsBigInt; amount: JsBigInt) {.importjs: "([#][0][0] += #)".} =
```

(until https://github.com/nim-lang/RFCs/issues/315 is resolved)